internal/engine/worker/decentralized/contract/polymarket/worker.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/rss3-network/node/config"
+	"github.com/rss3-network/node/internal/engine"
+	source "github.com/rss3-network/node/internal/engine/source/ethereum"
+	"github.com/rss3-network/node/provider/ethereum"
+	"github.com/rss3-network/node/provider/ethereum/contract"
+	"github.com/rss3-network/node/provider/ethereum/contract/polymarket"
+	"github.com/rss3-network/node/provider/ethereum/token"
+	"github.com/rss3-network/node/schema/worker/decentralized"
+	"github.com/rss3-network/protocol-go/schema"
+	activityx "github.com/rss3-network/protocol-go/schema/activity"
+	"github.com/rss3-network/protocol-go/schema/metadata"
+	"github.com/rss3-network/protocol-go/schema/network"
+	"github.com/rss3-network/protocol-go/schema/tag"
+	"github.com/rss3-network/protocol-go/schema/typex"
+	"github.com/samber/lo"
+	"github.com/shopspring/decimal"
+	"go.uber.org/zap"
+)
+
+var _ engine.Worker = (*worker)(nil)
+
+type worker struct {
+	ethereumClient ethereum.Client
+	tokenClient    token.Client
+	ctfExchange    *polymarket.CTFExchangeFilterer
+	// negRiskCTF     *polymarket.NegRiskCTFExchangeFilterer
+}
+
+func (w *worker) Name() string {
+	return decentralized.Polymarket.String()
+}
+
+func (w *worker) Platform() string {
+	return decentralized.PlatformPolymarket.String()
+}
+
+func (w *worker) Network() []network.Network {
+	return []network.Network{
+		network.Polygon,
+	}
+}
+
+func (w *worker) Tags() []tag.Tag {
+	return []tag.Tag{
+		tag.Collectible,
+	}
+}
+
+func (w *worker) Types() []schema.Type {
+	return []schema.Type{
+		typex.CollectibleTrade,
+	}
+}
+
+func (w *worker) Filter() engine.DataSourceFilter {
+	return &source.Filter{
+		LogAddresses: []common.Address{
+			polymarket.AddressPolyMarketCTFExchange,
+			polymarket.AddressPolyMarketNegRiskCTFExchange,
+		},
+		LogTopics: []common.Hash{
+			polymarket.EventOrderFinalized,
+		},
+	}
+}
+
+func (w *worker) Transform(ctx context.Context, task engine.Task) (*activityx.Activity, error) {
+	polygonTask, ok := task.(*source.Task)
+	if !ok {
+		return nil, fmt.Errorf("invalid task type %T", task)
+	}
+
+	activity, err := task.BuildActivity(activityx.WithActivityPlatform(w.Platform()))
+	if err != nil {
+		return nil, fmt.Errorf("build activity: %w", err)
+	}
+
+	activity.Type = typex.CollectibleTrade
+	activity.Actions = w.transformOrderTransaction(ctx, polygonTask)
+
+	return activity, nil
+}
+
+func (w *worker) transformOrderTransaction(ctx context.Context, polygonTask *source.Task) (actions []*activityx.Action) {
+	for _, log := range polygonTask.Receipt.Logs {
+		if len(log.Topics) == 0 {
+			continue
+		}
+
+		var (
+			buffer []*activityx.Action
+			err    error
+		)
+
+		if !w.matchOrderFinalizedLog(polygonTask, log) {
+			continue
+		}
+
+		buffer, err = w.transformOrderFinalizedLog(ctx, polygonTask, log)
+		if err != nil {
+			zap.L().Warn("handle polymarket order transaction", zap.Error(err), zap.String("worker", w.Name()), zap.String("task", polygonTask.ID()))
+			continue
```

Comment:
we should return errors when transform logs failed instead of skipping it, also use switch case to return the error once.